File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03807/p03807_num3_parsed.py
def solve(N, nums):
    # Count the number of even and odd integers
    even_count = 0
    odd_count = 0
    for i in range(N):
        if nums[i] % 2 == 0:
            even_count += 1
        else:
            odd_count += 1
    
    # If either even_count or odd_count is odd, it is possible to have only one integer on the blackboard
    if even_count % 2 == 1 or odd_count % 2 == 1:
        print("YES")
    else:
        print("NO")


# Read input values
N = int(input())
nums = list(map(int, input().split()))

# Solve the problem
solve(N, nums)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03807/p03807_num3_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N;
    cin >> N;

    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }

    int odd_count = 0;
    int even_count = 0;

    for (int i = 0; i < N; i++) {
        if (A[i] % 2 == 0) {
            even_count++;
        } else {
            odd_count++;
        }
    }

    if (odd_count % 2 == 1) {
        cout << "NO" << endl;
    } else {
        cout << "YES" << endl;
    }

    return 0;
}